src/io.d

```diff
@@ -68,11 +80,16 @@ struct Input
 
     Position position() const pure @safe
     {
-        import std.algorithm : find;
+        if (useOffset) {    
```

Comment:
Now, you can undo this `if`.
The extra work for the `text` argument is negligible.

Reply:
Maybe, the extra work on text argument is negligible. But the special offset mode becomes explicit. Without the `if` it will be implicit and hard to figure out when reading and therefore would asking for a explanation comment which could be avoided that way.
(I'm repeating my argumentation just in case you did not see it. I will not insist in that. Just say, "I prefer may way" :smiley:)

Reply:
If you write `if ` here then you should have two different `Position` constructors:
```
if (useOffset)
{
    return Position(name, col);
}
else
{
    ...
    return Position(name, line, col, text[begin .. end]);
}
```
Is this worth the extra effort?

Reply:
I agree now. The offset semantic should be explicit enough with the code and comments in the Input constructor:
```
    this(string name, const(char)[] text, bool useOffset = false) @nogc nothrow
    {
        ...
        this.useOffset = useOffset;
        if (useOffset) 
        {
            this.line = 0;
            this.col = 0;
        }
        else 
        {
            this.line = 1;
            this.col = 1;
        }
    }
```